h/queue.py

```diff
@@ -11,6 +13,8 @@ def __init__(self, namespace, *args, **kwargs):
     def publish(self, topic, data):
         if self.namespace is not None:
             topic = '{0}-{1}'.format(self.namespace, topic)
+        if not isinstance(data, str):
```

Comment:
If we're using JSON for message bodies everywhere then I think it would be easier to do both serialization and de-serialization in the queue. The motivation was to make it easier to mock `publish()` in testing, since 'data' can now be an dict where the values may be `mock.Mock` instances - if the caller has to do de-(serialization) then it runs into problems trying to JSON serialize dicts containing `mock.Mock` instances in tests.